Remotion/Web/UnitTests/Core/WebStringTest.cs

```diff
@@ -0,0 +1,210 @@
+    {
+      var stringValue = "My <b>test</a> string";
+      var webString = WebString.CreateFromString (stringValue);
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Text));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void CreateFromString_WithNullValue_ThrowsArgumentNullException ()
+    {
+      Assert.That (() => WebString.CreateFromString (null), Throws.ArgumentNullException);
+    }
+
+    [Test]
+    public void ImplicitConversionToString ()
+    {
+      var stringValue = "My <b>test</a> string";
+      WebString webString = stringValue;
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Text));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void IsEmpty ()
+    {
+      Assert.That (new WebString().IsEmpty, Is.True);
+      Assert.That (((WebString)"").IsEmpty, Is.True);
+      Assert.That (((WebString)" ").IsEmpty, Is.False);
+      Assert.That (((WebString)"test").IsEmpty, Is.False);
+    }
+
+    [Test]
+    public void Write_Text ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromString (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (webString.Write);
+      Assert.That (renderedString, Is.EqualTo ("My &lt;b&gt;test&lt;/a&gt;"));
+    }
+
+    [Test]
+    public void Write_Html ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromRawHtml (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (webString.Write);
+      Assert.That (renderedString, Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void Write_Attribute_ThrowsInvalidOperationException ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromRawAttribute (stringValue);
+
+      Assert.That (() => ExecuteWithHtmlTextWriter (webString.Write), Throws.InvalidOperationException);
+    }
+
+    [Test]
+    public void AddAttribute_TextWithAttributeEnum ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromString (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (
+          writer =>
+          {
+            webString.AddAttribute (writer, HtmlTextWriterAttribute.Class);
+            writer.RenderBeginTag ("a");
+            writer.WriteEndTag ("a");
+          });
+      Assert.That (renderedString, Is.EqualTo (@"<a class=""My &lt;b>test&lt;/a>""></a>"));
+    }
+
+    [Test]
+    public void AddAttribute_HtmlWithAttributeEnum_ThrowsInvalidOperationException ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromRawHtml (stringValue);
+
+      Func<string> testLambda = () => ExecuteWithHtmlTextWriter (writer => webString.AddAttribute (writer, HtmlTextWriterAttribute.Class));
+      Assert.That (testLambda, Throws.InvalidOperationException);
```

Comment:
we never have the test func outside of the Assert,That.

Also, I would prefer to have only one operation in the test func instead of the entire helper logic.